Components/PetFilters.jsx

```diff
@@ -163,8 +166,36 @@ export default function PetFilters({
       })
       filters += `],`
     }
+    let profBonusFilter = []
+    let profBonusFilter1 = ``
+    let profBonusFilter2 = ``
     if (profBonus[0] > 1 || profBonus[1] < 3) {
-      filters += `profBonus_in:[${GetBonusValues(profBonus[0], profBonus[1])}],`
+      profBonusFilter1 = GetBonusValues(profBonus[0], profBonus[1])
+    }
+    if (profBonusName.length > 0) {
+      profBonusName.forEach((c, i) => {
+        profBonusFilter2 += `${c.value}`
+        if (i < profBonusName.length - 1) {
+          profBonusFilter2 += `,`
+        }
+      })
+    }
+    if ((profBonusFilter1.length > 0) && (profBonusFilter2.length > 0)) {
+      // Merge where both values exist
+      profBonusFilter = profBonusFilter1.split(",").filter(v => profBonusFilter2.split(",").indexOf(v) > -1)
+      // if none match set to an empty query
+      if (profBonusFilter.length == 0) {
+        profBonusFilter = [ 0 ]
```

Comment:
Why not set it to [] here?

Reply:
This was kind of janky for sure.  An empty filter will return all results. In this case, there is no overlap between the 2 prof bonus filters.  An example would be selecting 1 star + innate greenskeeper (a 3 star only bonus).  These 2 are incompatible, so I set the filter value to 0 which won't match anything and the user gets no results instead of all results.

Reply:
This was kind of janky for sure. An empty filter will return all results. In this case, there is no overlap between the 2 prof bonus filters. An example would be selecting 1 star + innate greenskeeper (a 3 star only bonus). These 2 are incompatible, so I set the filter value to 0 which won't match anything and the user gets no results instead of all results.

Reply:
Alright, if you think it's reasonable. Haven't checked actual functionality. Just doublechecking.